### PR TITLE
Rust: Exclude data flow inconsistencies that stem from other inconsis…

### PR DESCRIPTION
**Pull request by paldepind**
Exclude some data flow inconsistency errors where the root cause is not in the data flow library.

Adds a new AST consistency metric that tracks AST nodes without a location.

I'm not sure if the change in this PR alone is enough for the new metric to show up on DCA. @geoffw0 most likely knows if additional setup is needed.

### CODE REVIEW

**Comment by geoffw0** [comment]
> I'm not sure if the change in this PR alone is enough for the new metric to show up on DCA

DCA pulls data from the `rust/summary/summary-statistics` query.  I believe you need to update the `getTotalDataFlowInconsistencies` predicate in `src\queries\summary\Stats.qll` which it uses - changes similar to what you did in `DataFlowConsistencyCounts.ql` should do it.

Otherwise LGTM.

File: rust/ql/consistency-queries/DataFlowConsistency.ql [hunk 1]
```diff
@@ -5,12 +5,4 @@
...
-import codeql.rust.dataflow.DataFlow::DataFlow as DataFlow
-private import rust
-private import codeql.rust.dataflow.internal.DataFlowImpl
-private import codeql.rust.dataflow.internal.TaintTrackingImpl
-private import codeql.dataflow.internal.DataFlowImplConsistency
-
-private module Input implements InputSig<Location, RustDataFlow> { }
-
-import MakeConsistency<Location, RustDataFlow, RustTaintTracking, Input>
+import codeql.rust.dataflow.internal.DataFlowConsistency
```
Comment (geoffw0) [inline rust/ql/consistency-queries/DataFlowConsistency.ql:8]: Adding this file has got rid of some needless duplication.  :+1:

**Comment by paldepind** [comment]
Thanks for the guidance @geoffw0. I missed the `Stats.qll`. I think that's why the DCA run didn't show any change in data flow inconsistencies. I've updated that as well and started a new run.

**Comment by paldepind** [comment]
DCA results are now looking as expected: Almost all data flow inconsistencies are gone and AST inconsistencies are up.

**Review by geoffw0 — APPROVED** [review]
Notably we now have more new AST inconsistencies than we've lost data flow inconsistencies, but I think that's expected and reflects where the problem really is.  And we have over 5 _million_ AST inconsistencies on `windows-rs`, which is spectacular, but nothing appears to have broken as a result.

It would [obviously] be good to prioritize fixing a majority of those AST inconsistencies fairly soon.

:+1:

**Review by hvitved — APPROVED** [review]

**Comment by paldepind** [comment]
Just to note, after this PR I think we want to keep an eye on the data flow inconsistencies in DCA and make sure that it doesn't go up. The DCA projects (not including Rust) all sits at 0 or 1 inconsistencies, so increases should be noticeable.

**Pull request merged**